### ISSUE TITLE
Reproduction of results

### ISSUE THREAD

**Issue by prion502** [MarcLafon/gallop#1]
Question 1: What is the epoch run for each dataset when in the case of 1shot, 2shot, etc. When I run 200 epochs in the eurosat dataset with the 1shot seed1 setting, the accuracy is only about 50, which is a very big difference from the results of your REPORT.

Question 2: The Food101 dataset at 16shot gives a result of 86.5 for table1 in the main text REPORT, yet it is 88 in the appendix Fig12.

Could the author please answer my question and give some specific implementation settings to easily reproduce the results?

**Comment by WuHuaGuoBoos** [comment]
> 问题 1：在 1shot、2shot 等情况下，每个数据集的 epoch 运行是多少。当我使用 1shot seed1 设置在 eurosat 数据集中运行 200 个 epoch 时，准确率只有 50 左右，这与你的 REPORT 结果有很大的差异。
> 
> 问题 2：在 16shot 处的 Food101 数据集在正文 REPORT 中给出的 table1 的结果为 86.5，但在附录 Fig12 中是 88。
> 
> 作者能否回答我的问题并给出一些具体的实现设置以轻松重现结果？

I'm in the same situation as you, and for now, I'd be grateful if you could recur

**Comment by WuHuaGuoBoos** [comment]
> 问题 1：在 1shot、2shot 等情况下，每个数据集的 epoch 运行是多少。当我使用 1shot seed1 设置在 eurosat 数据集中运行 200 个 epoch 时，准确率只有 50 左右，这与你的 REPORT 结果有很大的差异。
> 
> 问题 2：在 16shot 处的 Food101 数据集在正文 REPORT 中给出的 table1 的结果为 86.5，但在附录 Fig12 中是 88。
> 
> 作者能否回答我的问题并给出一些具体的实现设置以轻松重现结果？
I have encountered the same situation as you. Currently, have you reproduced the results? If so, I would be very grateful.

**Comment by WuHuaGuoBoos** [comment]
> 问题 1：在 1shot、2shot 等情况下，每个数据集的 epoch 运行是多少。当我使用 1shot seed1 设置在 eurosat 数据集中运行 200 个 epoch 时，准确率只有 50 左右，这与你的 REPORT 结果有很大的差异。
> 
> 问题 2：在 16shot 处的 Food101 数据集在正文 REPORT 中给出的 table1 的结果为 86.5，但在附录 Fig12 中是 88。
> 
> 作者能否回答我的问题并给出一些具体的实现设置以轻松重现结果？

I have encountered the same situation as you. Currently, have you reproduced the results? If so, I would be very grateful. ...

**Comment by WuHuaGuoBoos** [comment]
> 问题 1：在 1shot、2shot 等情况下，每个数据集的 epoch 运行是多少。当我使用 1shot seed1 设置在 eurosat 数据集中运行 200 个 epoch 时，准确率只有 50 左右，这与你的 REPORT 结果有很大的差异。
> 
> 问题 2：在 16shot 处的 Food101 数据集在正文 REPORT 中给出的 table1 的结果为 86.5，但在附录 Fig12 中是 88。
> 
> 作者能否回答我的问题并给出一些具体的实现设置以轻松重现结果？

I have encountered the same situation as you. Currently, have you replicated the results? If you could share, I would be very grateful.

**Comment by MarcLafon** [comment]
Hello @prion502,

Sorry for the late reply. When training with very low shots, we indeed use fewer training epochs to avoid overfitting, as done in previous works. Here are the number of epochs we used for each number of shots:

- 1 shot: 50 epochs,
- 2 and 4 shots: 100 epochs,
-  8 and 16 shots: 200 epochs.

In addition, the batch sizes are different for each dataset (depending on the number of classes in the dataset). Specifically for Eurosat we used a batch size of 10, and for Food101 we used a batch size 32.

I hope that these elements will help reproduce the results reported in the paper. 

Best,

Marc
